imap/ctl_mboxlist.c

```diff
@@ -562,71 +563,63 @@ static void do_dump(enum mboxop op, const char *part, int purge, int intermediar
 
 static void do_undump(void)
 {
-    int r = 0;
     char buf[16384];
     int line = 0;
-    const char *name, *partition, *acl;
-    int mbtype;
-    char *p;
 
     while (fgets(buf, sizeof(buf), stdin)) {
-        mbentry_t *newmbentry = NULL;
-        const char *server = NULL;
-
+        mbentry_t *newmbentry = mboxlist_entry_create();
         line++;
 
-        name = buf;
-        for (p = buf; *p && *p != '\t'; p++) ;
-        if (!*p) {
+        sscanf(buf, "%m[^\t]\t%d %ms %m[^>]>%ms " TIME_T_FMT " %" PRIu32
```

Comment:
I almost never use the `scanf` functions, so I'm reading up on them...

Looks like there's a set of `SCNu32` (etc) macros that are supposed to be used for `scanf`, and the `PRI...` ones are specifically for `printf` only.

I guess this is a portability thing, for handling platforms whose scanf/printf aren't consistent.  I don't know if such platforms exist, but it seems like an easy enough accommodation, so can you change the `PRIu32` here to `SCNu32` please?  Same applies for the other `sscanf` at 590.

I wonder if we'll run into problems with TIME_T_FMT?  But I guess, if we do, we can just fix them once we know what they are.

Reply:
As far as I can see the difference between PRIxxx and SCNxxx is the latter has `h` or `hh` prefix sometimes.  In case of PRIu32/SCNu32 the strings in inttypes.h are identical.  I have changed in 2× scanf PRIu32 ⇒ SCNu32.